ui/spinner.go

```diff
@@ -0,0 +1,47 @@
+package ui
+
+import (
+  "time"
+
+  "github.com/briandowns/spinner"
+)
+
+type Spinner struct{
+  spinner *spinner.Spinner
+  text string
+}
+
+func NewSpinner(text string) *Spinner { return defUI.NewSpinner(text) }
```

Comment:
I think this function and the one below should be in the `ui/ui.go` file as thats where the `UI` struct and subsequent global `defaultUI` are defined. Then this file would only contain the spinner struct itself and it's subsequent methods.